common/src/format.rs

```diff
@@ -338,7 +349,8 @@ impl FormatSpec {
         if pad_cnt > 0 {
             // separate with 0 padding
             let sep_cnt = disp_digit_cnt / (inter + 1);
-            let padding = "0".repeat((pad_cnt - sep_cnt) as usize);
+            let padding_char = if use_whitespace_padding { ' ' } else { '0' };
+            let padding = FormatSpec::compute_fill_string(padding_char, pad_cnt - sep_cnt);
```

Comment:
`use_whitespace_padding` could alternatively be an enum that `into`'s a `char`. A nice side-effect is that code would definitely read better (seeing `false` being supplied as an arg is mysterious as opposed to e.g `Padding::Zero` or `Padding::WhiteSpace`).

Reply:
```rust
#[derive(Debug)]
#[repr(u8)]
enum PaddingStyle {
    Null = b'\0',
    Whitespace = b' ',
}

impl Into<char> for PaddingStyle {
    fn into(self) -> char {
        self as u8 as char
    }
}
```
hard-coded padding has been extracted with `PaddingStyle` enum type. thanks for your advice :)

Reply:
Based on your review, extracted the hard-coded padding as a 'PaddingStyle' enum type
```rust
#[derive(Debug)]
#[repr(u8)]
enum PaddingStyle {
    Whitespace = b' ',
    Zero = b'0',
}

impl From<PaddingStyle> for char {
    fn from(from: PaddingStyle) -> char {
        from as u8 as char
    }
}
```

Reply:
ah, I should clarify a bit more now that I see your change. The idea behind the enum is to also change the function signatures to accept it, i.e `use_whitespace_padding` should just just be a `PaddingStyle` argument, so, for example:

```rust
fn separate_integer(
    magnitude_str: String,
    inter: i32,
    sep: char,
    disp_digit_cnt: i32,
    use_whitespace_padding: bool,
) -> String {
```

becomes:

```rust
fn separate_integer(
    magnitude_str: String,
    inter: i32,
    sep: char,
    disp_digit_cnt: i32,
    padding: PaddingStyle,
) -> String {
```

and call sites are changed from:

```rust
self.add_magnitude_separators(raw_magnitude_str?, sign_str, false);
```

into:

```rust
self.add_magnitude_separators(raw_magnitude_str?, sign_str, PaddingStyle::Zero);
```

In addition to these, the `if` statement that creates the `padding_char` can just be:

```rust
let padding_char = padding.into();
```

Reply:
ah I got it. I wasn't familiar with using Enum yet, so I couldn't handle it neatly. thanks for the feedback